packages/react-charts/src/components/Chart/Chart.tsx

```diff
@@ -527,6 +534,20 @@ export const Chart: React.FunctionComponent<ChartProps> = ({
     ...(name && { name: `${name}-${(legendComponent as any).type.displayName}` }),
     orientation: legendOrientation,
     theme,
+    ...(legendDirection === 'rtl' && {
+      dataComponent: legendComponent.props.dataComponent ? (
+        React.cloneElement(legendComponent.props.dataComponent, { transform: 'translate(40)' })
+      ) : (
+        <ChartPoint transform="translate(40)" />
+      )
+    }),
+    ...(legendDirection === 'rtl' && {
+      labelComponent: legendComponent.props.labelComponent ? (
+        React.cloneElement(legendComponent.props.labelComponent, { direction: 'rtl', dx: '10' })
+      ) : (
+        <ChartLabel direction="rtl" dx="10" />
+      )
+    }),
```

Comment:
The value of the `translate` prop isn't dependent on the text length, it's moving the `Point` a set distance from where the right aligned text ends which is static. There is no in-built "rtl" position for the `Point`, and without the translate the icons would show inside the text because it's still positioned relative to the text as "ltr".  

The text issue shown in the screenshot could be resolved by adjusting the padding, which the  `legendPosition` prop description kind of mentions but I think it's not for the same reason. I'll dig into the position calc logic - my thinking is that this is where we need to take the text length into account for rtl.

Reply:
@dlabrecq Updated the logic to account for text length for RTL legends, for very long text lengths you'll still need to adjust padding but that is what the position prop refers to being required. 

I tried both ways of adjusting the legend position and updating the translate ended up being simpler with less piping required for props. The extra `30` in the label offset is to shift the text further over to push the icon out to the right properly.

Reply:
For some reason, this was not working in the preview, so my thinking was that we needed to include the `legendXOffset`, like 

`translate(40 + ${legendXOffset})`

That said, it seems to be working for me locally.

Reply:
One more question... Why does the clone get `translate(40)`, but the non-clone gets `translate(legendXOffset)`?

![Screenshot 2023-09-21 at 5 03 55 PM](https://github.com/patternfly/patternfly-react/assets/17481322/c5ec1447-f6d0-4b9d-852b-6d4a2c52b734)

Reply:
Oh forgot to update those, will fix that. Perhaps updating the branch again will fix the preview too